### PR TITLE
GUI middle part of the HUD

### PR DESCRIPTION
**Pull request by ChipmunkV**
Rendering some paper.

### CODE REVIEW

**Comment by ChipmunkV** [comment]
~~Solved~~ Hacked around the transparency issue. So, the paper class has enough flexibility for the edges and scales properly.

File: assets/textures/paper.docx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# struct subtexture
+# one sprite, as part of a texture atlas.
+# 
+# this struct stores information about positions and sizes
+# of sprites included in the 'big texture'.
```
Comment (TheJJ) [inline assets/textures/paper.docx]: dunno if this comment is helpful :)

File: assets/textures/torn_paper_edge.docx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# struct subtexture
+# one sprite, as part of a texture atlas.
+# 
+# this struct stores information about positions and sizes
+# of sprites included in the 'big texture'.
```
Comment (TheJJ) [inline assets/textures/torn_paper_edge.docx]: same here

**Comment by TheJJ** [comment]
We also need the graphicaleffects package in the dependency matrix (in that `core, quick, quickcontrols ...` listing).

I couldn't try it yet, but why do we need the paper texture? Isn't something suitable in the original data?

Nevertheless, we should soon merge your awesome gui stuff, the diff is getting bigger and bigger :)

File: assets/textures/paper.docx (hunk 1 above)
Comment (ChipmunkV) [inline assets/textures/paper.docx]: Copy-pasted from other generated docx. Probably, only the 'x,y,w,h,cx,cy' comment line is useful.

**Comment by ChipmunkV** [comment]
In the original data there is a piece of paper of a fixed size. Not tileable. The HD version uses solid color for paper. Maybe there was a suitable paper texture from some other thing, but I looked only at the HUD ones.

**Comment by TheJJ** [comment]
rebase pls :)

**Comment by ChipmunkV** [comment]
Done, you might want to review the recent changes to the convert scripts here.

File: openage/convert/driver.py [hunk 3]
```diff
@@ -222,6 +223,39 @@ def convert_metadata(args):
...
+        filename = b'/'.join(filepath.parts).decode()
+        if filename.endswith(suffix):
+            try:
+                for line in filepath.open():
+                    match = re.match(r"\w+_files\s+(\w+)\s+\w+\s+(\w+)", line)
+                    if match:
+                        groups = match.group(2, 1)
+                        names_map[groups[0]] = groups[1].lower()
+            except UnicodeDecodeError:
+                continue
```
Comment (TheJJ) [inline openage/convert/driver.py]: why ignore the errors? i think we can avoid them alltogether if we `open(..)` them in `ISO-8859-1` or whatever microsoft used for the encoding.

Comment (ChipmunkV) [inline openage/convert/driver.py]: I don't think that these are all text files.

**Comment by TheJJ** [comment]
Looks good so far.
I'm a bit confused by the shaders in qml, which will pin us to OpenGL probably forever, but so be it.

It seems that the file renames require a reconversion for interface graphics, right? You need to add that to `changelog.py`.

**Comment by ChipmunkV** [comment]
Updated the `changelog.py`.

File: openage/convert/driver.py [hunk 4]
```diff
@@ -222,18 +223,55 @@ def convert_metadata(args):
...
+        ignored.update([('graphics', '.slp'), ('terrain', '.slp')])
+    if args.flag("no_interface"):
+        ignored.add(('interface', '.slp'))
 
     files_to_convert = []
     for dirname in ['sounds', 'graphics', 'terrain', 'interface']:
         for filepath in args.srcdir[dirname].iterdir():
-            if filepath.suffix in ignored_suffixes:
+            if (((None, filepath.suffix) in ignored) or
+                    ((filepath.parts[0].decode(), filepath.suffix) in ignored)):
```
Comment (TheJJ) [inline openage/convert/driver.py]: why do you decode here? isn't it a `str` already?

Comment (ChipmunkV) [inline openage/convert/driver.py]: No idea. I've taken that from the `path.py`:
``` 
   def name(self):
        """ The name of the topmost component (str). """
        return self.parts[-1].decode()
```

**Comment by ChipmunkV** [comment]
Forgot to comment: removed that texture file,

**Comment by TheJJ** [comment]
Kay :) How did you replace it, i thought it was necessary?

File: openage/convert/driver.py (hunk 4 above)
Comment (TheJJ) [inline openage/convert/driver.py]: seems alright then as you access internal components.
but the `parts` thingy is wrong itself i think.
``` python
>>> from pathlib import Path
>>> a = Path("/usr/local/bin/rofl/stuff")
>>> a.parts[0]
'/'
>>> a.parts
('/', 'usr', 'local', 'bin', 'rofl', 'stuff')
```

File: libopenage/gui/qml/Paper.qml [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+
+import QtQuick 2.4
+
+Item {
+	id: root
+
+	/*
+	 * A paper texture.
+	 */
+	property url paperTextureSource: "image://by-filename/converted/interface/scr5b.slp.png.0"
```
Comment (ChipmunkV) [inline libopenage/gui/qml/Paper.qml]: That's a texture.

**Comment by ChipmunkV** [comment]
Made the paper edge mask black.

About the shaders: the opacity shader won't be needed in Qt 5.7, the other one is the precise kind of burn effect that I wanted.

File: openage/convert/interface/interfacecutter.py [hunk 6]
```diff
@@ -24,34 +27,42 @@ def __call__(self, image):
...
+                                   MID_STRIP_PATTERN_CORNERS,
+                                   MID_STRIP_PATTERN_SEARCH_AREA_CORNERS)
 
             for coords in KNOWN_SUBTEX_CORNER_COORDS:
                 yield TextureImage(pil_image.crop(coords))
         else:
             yield image
 
-    def __cut_top_strip(self, pil_image):
+    def __cut_strip(self, pil_image, pattern_corners, search_area_corners):
```
Comment (TheJJ) [inline openage/convert/interface/interfacecutter.py]: hm i think we can remove the `__`, and if you really wanna keep it, a single `_` should be enough.

**Comment by TheJJ** [comment]
Looks really cool!
openage_2016-06-01_15-19-51_00

File: openage/convert/interface/interfacecutter.py (hunk 6 above)
Comment (ChipmunkV) [inline openage/convert/interface/interfacecutter.py]: Ok, and I'll fix the comment, so it's not specifically about the top strip.

**Comment by TheJJ** [comment]
Awesome! Woooh!

**Pull request merged**